src/traces/waterfall/attributes.js

```diff
@@ -0,0 +1,68 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var extendFlat = require('../../lib/extend').extendFlat;
+
+var barAttrs = require('../bar/attributes');
+
+module.exports = {
+
+    x: barAttrs.x,
+    x0: barAttrs.x0,
+    dx: barAttrs.dx,
+    y: barAttrs.y,
+    y0: barAttrs.y0,
+    dy: barAttrs.dy,
+
+    r: barAttrs.r,
+    t: barAttrs.t,
+
+    text: barAttrs.text,
+    hovertext: barAttrs.hovertext,
+    hovertemplate: barAttrs.hovertemplate,
+
+    textposition: barAttrs.textposition,
+
+    textfont: barAttrs.textfont,
+
+    insidetextfont: barAttrs.insidetextfont,
+
+    outsidetextfont: barAttrs.outsidetextfont,
+
+    constraintext: barAttrs.constraintext,
+
+    cliponaxis: barAttrs.cliponaxis,
+
+    orientation: barAttrs.orientation,
+
+    base: barAttrs.base,
+
+    offset: barAttrs.offset,
+
+    width: barAttrs.width,
+
+    marker: extendFlat({}, barAttrs.marker, {
```

Comment:
We should have two `marker` containers for waterfall traces: one for increasing bars and one for decreasing bars, similar to how `ohlc` and `candlestick` do it.

https://github.com/plotly/plotly.js/blob/02e51d8f5ba5601f53ee3dc839ba05fb12c7f8a2/src/traces/ohlc/attributes.js#L89-L91